src/tbb/arena.h

```diff
@@ -552,43 +549,32 @@ void arena::advertise_new_work() {
     }
 }
 
-inline long long arena::get_index_for_steal(FastRandom& frnd, std::size_t slot_num_limit) {
-    std::size_t k = std::count_if(pool_mask, pool_mask + slot_num_limit, [] (std::atomic<int>& el) {
-        return el.load(std::memory_order_relaxed);
-    });
-
-    if (k == 0) {
-        return -1;
-    }
+inline d1::task* arena::steal_task(FastRandom& frnd, execution_data_ext& ed, isolation_type isolation) {
+    auto get_index_for_steal = [&] {
+        std::size_t k = std::count_if(pool_mask, pool_mask + my_num_slots, [] (std::atomic<int>& el) {
+            return el.load(std::memory_order_relaxed);
+        });
+        if (k == 0) {
+            return -1;
+        }
 
-    std::size_t target_slot = (frnd.get() % k) + 1;
-    std::size_t i = 0; 
-    for (; target_slot > 0 && i < slot_num_limit; ++i) {
-        if (pool_mask[i].load(std::memory_order_relaxed)) {
-            --target_slot;
+        std::size_t target_slot = frnd.get() % k;
+        for (unsigned i = 0; i < my_num_slots; ++i) {
+            if (pool_mask[i].load(std::memory_order_relaxed) && target_slot-- == 0) {
```

Comment:
Readability of this code line is questionable. I would recommend not to create this conditions where decrement is shadowed.